dotnet/src/SemanticKernel/Planning/IPlan.cs

```diff
@@ -0,0 +1,84 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Collections.Generic;
+using System.Text.Json.Serialization;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.SemanticKernel.Orchestration;
+
+namespace Microsoft.SemanticKernel.Planning;
+
+public interface IPlan
+{
+    /// <summary>
+    /// Goal of the plan
+    /// </summary>
+    string Goal { get; }
```

Comment:
why not IPlan?

Reply:
I don't like IPlan, hard to know where the actual goal is. I do like the idea of merging with PlanStep.